### PR TITLE
[shaderc] Re-fix installation

### PR DESCRIPTION
**Pull request by JackBoosY**
- Enable dynamic build and re-fix the dependency relationship.
- Remove the util installation patch since we don't need it (added in https://github.com/microsoft/vcpkg/pull/18979).
- Disable build sample.
- Re-factory portfile.cmake.

Related: #16137.

### CODE REVIEW

**Comment by JackBoosY** [comment]
@jmoguillansky-gpsw  Can you please test this PR?

Thanks.

**Comment by NancyLi1013** [comment]
Related issue #16658

**Review by NancyLi1013 — APPROVED** [review]

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
Please see https://github.com/microsoft/vcpkg/pull/16137#issuecomment-802952196

**Comment by jmoguillansky-gpsw** [comment]
> @jmoguillansky-gpsw Can you please test this PR?
> 
> Thanks.

Hi @JackBoosY I tested your branch, I was able to build shaderc shared lib successfully.
Thanks, much appreciated!

**Comment by JackBoosY** [comment]
@ras0219-msft glslang and spir-tools use some symbols that are not exported in shaderc, so if we need to build them, we must first build static shaderc.

**Comment by ras0219-msft** [comment]
>glslang and spir-tools use some symbols that are not exported in shaderc

shaderc _depends_ on glslang and spirv-tools; therefore they cannot be depending on symbols exported by shaderc:

```
# ports/control/shaderc

Build-Depends: glslang, spirv-tools
```

Therefore, if glslang and spirv-tools can only be built statically, shaderc should also be built statically. Dynamic libraries should (generally) not depend on static libraries.

**Comment by jmoguillansky-gpsw** [comment]
> The reason I need to use shaderc shared library is because visual studio toolchain doesn't let you mix /MT (multithreaded) and /MD (multithreaded DLL) so if I have a shared library that links against shaderc, it needs to link against the shared lib of shaderc if I understand correctly?

I see there's an option: "DSHADERC_ENABLE_SHARED_CRT", maybe it's possible to use the shaderc static lib if I set this option?
But I don't think vcpkg exposes this option, maybe I need to build shaderc manually from source?

**Comment by CaptainHorse** [comment]
Any updates on this? Would love to use shaderc thru vcpkg as it has issues with path size limits in my project.

**Comment by JackBoosY** [comment]
@CaptainHorse  Will continue to handle this PR later.

File: versions/s-/shaderc.json [hunk 1]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "acddbab84ab19cc4d1e04ecb5277d3969cf9b093",
```
Comment (JackBoosY) [inline versions/s-/shaderc.json:4]:
```suggestion
      "git-tree": "f6a82ca83afbf20c08d6d001f133cd5fd2672108",
```

**Comment by JackBoosY** [comment]
Requires https://github.com/microsoft/vcpkg/pull/19219.

**Comment by JackBoosY** [comment]
Linux regression related to https://github.com/KhronosGroup/SPIRV-Tools/issues/1569.

**Comment by PhoebeHui** [comment]
The related upstream issue on linux has been fixed, please reopen and continue this PR if you have time.

**Pull request closed without merging**